### PR TITLE
Site

### PR DESCRIPTION
**Pull request by joshplumley**
Site to ~~Main~~ planningQueue

### CODE REVIEW

**Review by jbolles94 — APPROVED** [review]

**Comment by jarrilla** [comment]
@joshplumley @jbolles94 is this ready to review ? tests seem to pass

File: src/site/controller.ts [hunk 1]
```diff
@@ -3,44 +3,187 @@
...
+      if (!name) {
+        return HTTPError('Missing required arg, name.', 400);
+      }
+
+      if (!location) {
+        return HTTPError('Missing required arg, location.', 400);
+      }
+
+      if (!timezone) {
+        return HTTPError('Missing required arg, timezone.', 400);
```
Comment (jarrilla) [inline src/site/controller.ts:83]: Is there a library we could use to check for valid timezone strings ?

File: src/site/controller.ts [hunk 2]
```diff
@@ -3,44 +3,187 @@
...
+import { SiteModel } from './model';
+
+const SiteRouter = Router();
+export default SiteRouter;
+
+SiteRouter.get('/', getAllSites);
+SiteRouter.put('/', createSite);
+SiteRouter.delete(
+  '/',
+  async (req, res, next) =>
```
Comment (jarrilla) [inline src/site/controller.ts:19]: All of these middlewares could be written as `(req, res, next) => checkId(res, next, ...),` due to the fact that `next` is what will let express know to continue to the endpoint handler. There is no need to write `async (req, res, next) => await f()`, as the `async/await` will have no effect.

**Review by jarrilla — CHANGES_REQUESTED** [review]
Tests look good and work fine. Couple of clean up items required.

File: src/site/controller.ts (hunk 1 above)
Comment (jbolles94) [reply src/site/controller.ts:83]: I was going to use Moment, but it is deprecated now. Instead changed to using the Intl object that is compatible with our versions of node. MR is out for it here: https://github.com/conturo-prototyping/pack-ship-server/pull/133

File: src/site/controller.ts (hunk 2 above)
Comment (jbolles94) [reply src/site/controller.ts]: Removed async/await from middleware

**Comment by jarrilla** [comment]
@jbolles94  "Should hold job" fails and bails. Let me know if this is the same for you.

**Pull request merged**